tests/v2/validation/registries/registry_test.go

```diff
@@ -0,0 +1,245 @@
+package registries
+
+import (
+	"testing"
+
+	"github.com/rancher/rancher/tests/framework/clients/corral"
+	"github.com/rancher/rancher/tests/framework/clients/rancher"
+	management "github.com/rancher/rancher/tests/framework/clients/rancher/generated/management/v3"
+	v1 "github.com/rancher/rancher/tests/framework/clients/rancher/v1"
+	"github.com/rancher/rancher/tests/framework/extensions/clusters"
+	"github.com/rancher/rancher/tests/framework/extensions/registries"
+	nodepools "github.com/rancher/rancher/tests/framework/extensions/rke1/nodepools"
+	"github.com/rancher/rancher/tests/framework/extensions/workloads/pods"
+	"github.com/rancher/rancher/tests/framework/pkg/config"
+	"github.com/rancher/rancher/tests/framework/pkg/environmentflag"
+	namegen "github.com/rancher/rancher/tests/framework/pkg/namegenerator"
+	"github.com/rancher/rancher/tests/framework/pkg/session"
+	"github.com/rancher/rancher/tests/framework/pkg/wait"
+	"github.com/rancher/rancher/tests/integration/pkg/defaults"
+	provisioning "github.com/rancher/rancher/tests/v2/validation/provisioning"
+	"github.com/rancher/rancher/tests/v2/validation/provisioning/rke1"
+	"github.com/sirupsen/logrus"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	"github.com/stretchr/testify/suite"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	corralRancherName      = "rancherha"
+	corralAuthDisabledName = "registryauthdisabled"
+	corralAuthEnabledName  = "registryauthenabled"
+)
+
+type RegistryTestSuite struct {
+	suite.Suite
+	session                        *session.Session
+	client                         *rancher.Client
+	podListClusterAuth             *v1.SteveCollection
+	podListClusterNoAuth           *v1.SteveCollection
+	podListClusterLocal            *v1.SteveCollection
+	clusterAuthRegistryHost        string
+	clusterNoAuthRegistryHost      string
+	localClusterGlobalRegistryHost string
+	rancherUsesRegistry            bool
+}
+
+func (rt *RegistryTestSuite) TearDownSuite() {
+	rt.session.Cleanup()
+}
+
+func (rt *RegistryTestSuite) SetupSuite() {
+	testSession := session.NewSession()
+	rt.session = testSession
+
+	corralConfig := corral.CorralConfigurations()
+	registriesConfig := new(Registries)
+	config.LoadConfig(RegistriesConfigKey, registriesConfig)
+
+	err := corral.SetupCorralConfig(corralConfig.CorralConfigVars, corralConfig.CorralConfigUser, corralConfig.CorralSSHPath)
+	require.NoError(rt.T(), err)
+	configPackage := corral.CorralPackagesConfig()
+
+	environmentFlags := environmentflag.NewEnvironmentFlags()
+	environmentflag.LoadEnvironmentFlags(environmentflag.ConfigurationFileKey, environmentFlags)
+	useRegistries := environmentFlags.GetValue(environmentflag.UseExistingRegistries)
```

Comment:
I think client instantiation needs to happen before these. Because these are already gonna happen once the client is instantiated by default.

Then we'd be able to do `client.Flags.GetValue` to get the value. And we wouldn't need flag instantiation (L64) and load (L65) in the tests.

Reply:
100% correct, and also the client instantiation should just be `client, err := rancher.NewClient("", testSession)` there is no need to load the rancher config, the client does it by default.